packages/front-end/pages/404.tsx

```diff
@@ -0,0 +1,37 @@
+import * as Sentry from "@sentry/react";
+import { useRouter } from "next/router";
+import { useEffect } from "react";
+import { isSentryEnabled } from "@/services/env";
+
+export default function Custom404() {
+  const ind = Math.ceil(Math.random() * 2);
+  const router = useRouter();
+  const badPath = router?.asPath;
+  useEffect(() => {
+    if (!isSentryEnabled()) return;
+    if (!badPath) return;
+
+    const referrer = document && document?.referrer ? document.referrer : "-";
```

Comment:
This can still throw a ReferenceError. In all cases there is a reference to document, it will be truthy. I'd insert a chceck before this, e.g.:

```ts
if (typeof document === 'undefined') return;

const referrer = document.referrer || '-'
```

Reply:
I know that sometimes referrer isn't defined, but didn't want it not to log a 404 in that case. Would you be okay if I wrapped the referrer in that typeof check?

Reply:
`const referrer = (typeof document === 'undefined') ? "-" : document?.referrer ? document.referrer : "-";`

Reply:
Yes, logically that works. Stylistically, if you wanted to avoid a nested ternary you could do this:

```ts
const referrer = (typeof document === 'undefined') ? "-" : document?.referrer || "-";
```